### PR TITLE
Saner way to add shared libraries for android.

### PR DESCRIPTION
**Pull request by Gleefre**
Uses `:android` flag and `android-<abi>-libmixed.so` libraries.

Previous way has two issues:
`lin-arm64-libmixed.so` was replaced by android library, but linux and android libraries are not compatible.
`libmixed.so` for `amd64` (x86-64) wasn't added. (It might be useful for running in emulator for example.)

### CODE REVIEW

File: low-level.lisp [hunk 1]
```diff
@@ -30,20 +36,17 @@
...
-        #+ARM64 "bsd-arm64-libmixed.so"
-        #+(and ARM (not ARM64)) "bsd-arm7a-libmixed.so"))
+        #+ARM64 "bsd-arm64-libmixed.so"))
   (:windows
    (:or "libmixed2.dll" "libmixed.dll"
         #+X86 "win-i686-libmixed.dll"
         #+X86-64 "win-amd64-libmixed.dll"
-        #+ARM64 "win-arm64-libmixed.dll"
-        #+(and ARM (not ARM64)) "win-arm7a-libmixed.so"))
+        #+ARM64 "win-arm64-libmixed.dll"))
```
Comment (Shinmera) [inline low-level.lisp:49]: Why did you remove these clauses though?

Comment (Gleefre) [reply low-level.lisp:49]: Hm, I didn't notice them, I just reverted the previous commit.

**Comment by Gleefre** [comment]
I added arm7a paths back.

**Comment by Shinmera** [comment]
Thanks again!

**Comment by Gleefre** [comment]
My pleasure :)
